sys/arm64/include/cheric.h

```diff
@@ -56,4 +56,8 @@ trapframe_set_elr(struct trapframe *tf, uintcap_t elr)
 }
 #endif
 
+#if __has_feature(capabilities)
+#define	cheri_capmode(cap)	(__typeof(cap))((uintcap_t)(cap) | 1)
```

Comment:
Interesting, I didn't know that `__typeof` also worked in addition to `__typeof__` (and of course `typeof` depending on compiler flags).

I do find cheri_capmode() sounds more like a boolean query than something that turns on capmode. Maybe `cheri_enable_capmode()`?

Reply:
Oops, I will use the more standard `__typeof__`.  I would be fine with the 'enable' name.

Reply:
I think I went with `__typeof` because it was widely used in `sys/` (102 for that vs only 21 for `__typeof__`).